packages/flutter/lib/src/material/text_field.dart

```diff
@@ -192,6 +195,11 @@ class TextField extends StatefulWidget {
   /// {@macro flutter.widgets.editableText.maxLines}
   final int maxLines;
 
+  /// If [maxLength] is set to this number, it disables the showing
```

Comment:
If [maxLength] is set to this value, only the "current input length" part of the character counter is shown.